README.md

```diff
@@ -145,6 +146,88 @@ export class NotFoundComponent {
 }
 ```
 
+## Providers
+
+```typescript
+export interface Token {
+  access: string;
+}
+
+export const getTokenFromCookie(req): Token {
+  // code to extract token with request object
+  const token = { access: 'sdfsdffasdsd' };
+  return token;
+}
+
+import { Module } from '@nestjs/common';
+import { join } from 'path';
+import { AngularUniversalModule } from '@nestjs/ng-universal';
+
+@Module({
+  imports: [
+    AngularUniversalModule.forRoot({
+      bootstrap: AppServerModule,
+      viewsPath: join(process.cwd(), 'dist/{APP_NAME}/browser'),
+      extraProviders: [{ provide: 'SERVER_TEST', useValue: 10 }],
+      dynamicProviders: [{ provide: 'SERVER_TOKEN', useValue: (req, res) => getTokenFromCookie(req) }]
```

Comment:
```suggestion
      dynamicProviders: [{ provide: 'SERVER_TOKEN', useValue: (req, res) => __HERE__ }]
```

Maybe we can just add an example inline code (see __HERE__) instead of using the `getTokenFromCookie` function

Reply:
```suggestion
      dynamicProviders: (req, res) => [{ provide: 'SERVER_TOKEN', useValue: { access: req.headers.get('token') } }]
```

Actually the usage is wrong based on how it's implemented: `ngOptions.dynamicProviders(options.req, options.res)`